vault/quotas/quotas.go

```diff
@@ -745,6 +755,21 @@ func (m *Manager) Reset() error {
 
 // Must be called with the lock held
 func (m *Manager) resetCache() error {
+	names, err := m.quotaNamesLocked(TypeRateLimit)
+	if err != nil {
+		return err
+	}
+	for _, name := range names {
+		quota, err := m.quotaByNameLocked(TypeRateLimit.String(), name)
+		if err != nil {
+			return err
+		}
+		rlq := quota.(*RateLimitQuota)
```

Comment:
can quotaByNameLocked return `nil, nil`? might want to do a `nil` check here just in case...

Reply:
You're right, it seems like it can! I never hit this in my testing but good spot. I'll add the nil check.